docs: add context in the spring getting started guide

pepopowitz opened the pull request:
## Description

Outcome of #5246.

Adds context to the Spring Getting Started guide. I noticed these things could use more information while stepping through the guide. 

I'm opening this PR against v8.8 to solicit feedback. Once I've received/addressed the feedback, I'll backport the changes to previous versions of the guide, as part of this PR.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [ ] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [x] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and:
  - [x] are in the `/docs` directory (version 8.8).
  - [ ] ‼️ are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [ ] ‼️ My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [x] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -36,7 +36,7 @@ While stepping through this guide, you can visit our [sample repository](https:/
 
 - Access to your preferred code editor or IDE
-- [OpenJDK 21+](https://openjdk.org/install/)
+- [OpenJDK 21+](https://openjdk.org/install/) and [Maven 3](https://maven.apache.org/index.html)
```
I installed the JDK differently this time (https://mise.jdx.dev/lang/java.html) than previous times running through this guide. I'm not sure if the requirement to install Maven has always been separate, or if it depends on how you install a JDK.

akeller replied:
Good add. I would assume a Java dev would have Maven, but it's not packaged with the JDK.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -72,7 +72,7 @@ Next, we'll create a BPMN diagram to represent the transaction model shown at th
 1. Open Desktop Modeler.
 2. Click **Create a new diagram** in Camunda 8.
-3. In the properties panel, under the **General** section:
+3. Toggle the properties panel from the Window menu if it is not already visible, and make the following changes in the **General** section:
```
I struggled to find the properties panel with a fresh install of Modeler 😅

akeller replied:
Can you take this feedback to the Desktop Modeler team? I'm not sure if it's collapsed by default.

pepopowitz replied:
I'm unclear what the feedback is. My experience with a fresh installation of Modeler is that _it is_ collapsed by default. Do you mean that it shouldn't be?

christinaausley replied:
I do not love this new UI -- the light gray on white is tough to clearly see.

Perhaps "If the properties panel is not already visible, expand it by clicking the gray toggle on the right side of the screen(or canvas?)."

pepopowitz replied:
Wow, I didn't even _see_ that light gray affordance until you said something. 

Do you have a strong preference for guiding them to the visual toggle vs. selecting something from the menu as my changes describe? My preference is the menu, but if you feel strongly that it should be about the visual toggle, I will change it.

christinaausley replied:
The menu is fine by me!

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -82,22 +82,23 @@ Next, we'll create a BPMN diagram to represent the transaction model shown at th
    3. **Script/FEEL expression**: `total * 1.1` (this represents the tax applied to the transaction.)
 7. Append a task named `Charge credit card`.
-8. Click on the task and click the wrench-shaped icon to change the type of task to a service task. In the properties panel, change the **Task definition/Type** to `charge-credit-card`.
+8. Click on the task and click the wrench-shaped icon to change the type of task to a service task. In the properties panel, change the **Task definition/Job type** to `charge-credit-card`.
```
This description probably changed since we wrote the guide, it now reads "Job type" in the UI of Desktop Modeler.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -82,22 +82,23 @@ Next, we'll create a BPMN diagram to represent the transaction model shown at th
 10. Save this BPMN file to your Spring project in `src/main/resources`, and name it `process-payments.bpmn`.
 
-## Step 3: Deploy your process
+## Step 3: Deploy your process from Desktop Modeler
```
We show them how to deploy the process twice in this guide, in two different ways.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -82,22 +82,23 @@ Next, we'll create a BPMN diagram to represent the transaction model shown at th
-2. Change the **Deployment name** to `process-payments`, and ensure the **Target** is `Camunda 8 Self-Managed`.
-3. Change the **Cluster endpoint** to `http://localhost:26500/`, with no authentication.
-4. Click **Deploy**.
+1. Open Desktop Modeler and ensure the desired Camunda8 version is targeted in the bottom left corner.
```
Especially with two different alphas available, this feels like an easy thing to mess up.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -82,22 +82,23 @@ Next, we'll create a BPMN diagram to represent the transaction model shown at th
+1. Open Desktop Modeler and ensure the desired Camunda8 version is targeted in the bottom left corner.
+2. Click the rocket icon in the bottom left corner.
+3. Change the **Deployment name** to `process-payments`, and ensure the **Target** is `Camunda 8 Self-Managed`.
+4. Change the **Cluster endpoint** to `http://localhost:26500`, with no authentication.
```
No slash at the end of the cluster endpoint URL, or else Modeler will tell you it can't connect to the cluster.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -113,7 +114,7 @@ To implement a service task, take the following steps:
 ### Configure Spring Boot Starter
 
-Add the following Maven dependency to your Spring Boot Starter project, replacing `x` with the latest patch level available:
+Add the following Maven dependency to your Spring Boot Starter project's pom.xml file, replacing `x` with [the latest patch level available](https://mvnrepository.com/artifact/io.camunda/spring-boot-starter-camunda-sdk):
```
1. `pom.xml`: I had a moment of panic as I forgot where the dependencies go in a spring project 😭.
2. I personally can't ever figure out what the latest patch level is, I wouldn't expect a reader to know either. Linking them to maven central seems like a quick way to figure that out.

akeller replied:
Ideally, we have some automated way to increment this instead of "x" but I like your addition of a link here for now.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -125,7 +126,8 @@ Add the following Maven dependency to your Spring Boot Starter project, replacin
 ### Configure the Camunda client
 
-Open your `src/main/resources/application.yaml` file, and paste the following snippet to connect to the Self-Managed Zeebe Broker:
+1. [Rename your `src/main/resources/application.properties` file to `application.yaml`](https://mvnrepository.com/artifact/io.camunda/spring-boot-starter-camunda-sdk).
```
The original download from start.spring.io is in the plain-text format (application.properties), so if we want them to configure via yaml, we should tell them to change that file type.

pepopowitz commented on docs/guides/getting-started-java-spring.md:
```diff
@@ -177,6 +179,8 @@ To check your work, visit our [sample repository](https://github.com/camunda/cam
 
 In your terminal, run `mvn spring-boot:run`, where you should see the `charging credit card` output. In Operate, refresh if needed, and note the payment has executed.
 
+Since the application is now running a worker subscribed to the workflow engine, it will not exit until you hit `Ctrl+C`.
```
There's no such thing as obvious, someone will surely be surprised that their app is no longer exiting on its own.

akeller replied:
Good add. 

Non-blocking feedback: Do we need to add something here for the Mac users? I can talk myself both in and out of it.

christinaausley reviewed (commented):
Nice -- a few minor comments, but this looks great!

akeller commented:
>I'm opening this PR against v8.8 to solicit feedback. Once I've received/addressed the feedback, I'll backport the changes to previous versions of the guide, as part of this PR.

He actually won't because he'll enable auto-merge and launch it to the moon immediately 😜

pepopowitz commented:
> > I'm opening this PR against v8.8 to solicit feedback. Once I've received/addressed the feedback, I'll backport the changes to previous versions of the guide, as part of this PR.
> 
> He actually won't because he'll enable auto-merge and launch it to the moon immediately 😜

oh noooooo oops! Fixing.
